[Feat] benefitCollectionView Section 1 구현 및 Section0,1 데이터구조 (#18)

sjk4618 opened the pull request:
## 🔥*Pull requests*

⛳️ **작업한 브랜치**
- #18

👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
- section 1 구현
- 네트워크에서 주는 데이터 구조체 생성 및 더미데이터 넣기
- section 0과 1에 데이터 뿌려주기

## 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
1. section 1의 backgroundView와 헤더뷰 중첩 문제
<img width="223" alt="스크린샷 2023-11-23 오전 6 27 49" src="https://github.com/SOPT-33RD-APP-NAVERPAY/NaverPay-iOS/assets/70939232/42798848-3503-4109-bf02-dcd615fd49e8">

위 뷰에서 backgroundView를 깔면 헤더부분까지 깔리는 문제가 생겼습니다.
헤더말고 하얀배경의 cell 부분의 conerradius를 주기가 어려웠고, 색깔도 바꿔줘야했습니다.
그래서 아래와 같이 backgroundView안에 view를 하나 더 만들어서 잡아줬습니다.

<img width="546" alt="스크린샷 2023-11-23 오전 6 30 06" src="https://github.com/SOPT-33RD-APP-NAVERPAY/NaverPay-iOS/assets/70939232/a750a9f9-6745-4ed8-9d1e-a89b47b20dba">              
    
  -------------

2. Label 길이 레이아웃 문제

<img width="1032" alt="스크린샷 2023-11-23 오전 5 47 09" src="https://github.com/SOPT-33RD-APP-NAVERPAY/NaverPay-iOS/assets/70939232/eb24c151-3765-4909-9221-b2c60ecfeac4">

위 뷰에서 네이버페이 옆 "회원은 매일"이 잘리는 문제가 발생하였습니다. 그래서 사진과 같이 lessthanOrEqualTo를 사용해서 해결했습니다.   
     
 -------------
    
3. NPLabel 컴포넌트 생성
```swift
import UIKit

final class NPLabel: UILabel {
    init(
        font: UIFont,
        color: UIColor
    ) {
        super.init(frame: .zero)
        self.font = font
        self.textColor = color
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

```
사용법은 제 코드 잘보시면 있습니다..! 
font와 color를 바로 쓸 수 있도록 만들어두었습니다!


## 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|GIF|![Simulator Screen Recording - iPhone 13 mini - 2023-11-23 at 06 24 14](https://github.com/SOPT-33RD-APP-NAVERPAY/NaverPay-iOS/assets/70939232/74e1b19f-7e32-40f3-a471-c9a6cbae9f01)|


## 📟 관련 이슈
- Resolved: #18

sjk4618 committed:
[Feat] BenefitCollectionView section 1 UI 구현(#18)

sjk4618 committed:
[Feat] 앱데이터 구조체 생성 및 섹션1 데이터 넘기기 구현(#18)

sjk4618 committed:
[Feat] FamousBenefitSection cell 구현(#18)

sjk4618 committed:
[Feat] section background cornerradius 조절(#18)

sjk4618 committed:
[Feat] benefitCollectionView Section1 UI 구현(#18)

sjk4618 committed:
[Feat] 구조체 데이터 전달 및 레이아웃 수정(#18)

sjk4618 committed:
[Chore] 이미지 변경(#18)

kim-seonwoo commented on NaverPay/Scenes/Benefit/Cells/BenefitCollectionViewFamousBenefitCell.swift:
```diff
@@ -0,0 +1,104 @@
+    
+    private let brandNameLabel = NPLabel(font: .font(.body_bold_15), color: .sub_black)
+    private let benefitDescriptionLabel = NPLabel(font: .font(.body_medium_15), color: .sub_black)
+    private let benefitRate = NPLabel(font: .font(.body_bold_16), color: .main_green)
```
NPlabel을 이런식으로 사용하면 되군요~~!

kim-seonwoo commented on NaverPay/Scenes/Benefit/Models/UserBenefitAppData.swift:
```diff
@@ -0,0 +1,59 @@
+extension UserBenefitDataAppData {
+    static func dummy() -> UserBenefitDataAppData {
+        
+            UserBenefitDataAppData(userName: "곽성준", userPoint: 11500, brandList: [
```
저도 이런식으로 더미 데이터 만들어 봐야겠습니다!

kim-seonwoo commented on NaverPay/Scenes/Benefit/Views/BenefitCollectionViewFamousBenefitSectionBackgroundView.swift:
```diff
@@ -0,0 +1,47 @@
+
+    }
+    
+    private func setLayout() {
```
저도 같은 문제를 겪어서 HeaderView영역이 Background View의 상의레이어로서 겹쳐서 덮도록 해결했었습니다.
이렇게 뷰를 분리하는게 더 깔끔하네요!

kim-seonwoo reviewed (commented):
NPLabel 사용해 보겠습니다 감사합니다~~

sjk4618 committed:
[Fix] Conflict 해결(#18)

# Conflicts:
#	NaverPay.xcodeproj/project.pbxproj
#	NaverPay/Scenes/Home/ViewControllers/HomeViewController.swift

sjk4618 committed:
[Chore] 파일 이동(#18)
